File: sc-demo-node-http/sdk/heartbeat.js
const fs = require("node:fs");
const path = require("node:path");

function writeHeartbeat(rootDir, status = "alive", extra = {}) {
  const filePath = path.join(rootDir, "runtime", "heartbeat.json");
  fs.mkdirSync(path.dirname(filePath), { recursive: true });
  const payload = {
    timestamp: new Date().toISOString(),
    status,
    ...extra,
  };
  fs.writeFileSync(filePath, JSON.stringify(payload, null, 2));
}

module.exports = { writeHeartbeat };

File: my-python-task/sdk/heartbeat.py
import json
import pathlib
from datetime import datetime, timezone


def write_heartbeat(root: str | pathlib.Path, status: str = "alive", extra: dict | None = None) -> None:
    root_path = pathlib.Path(root)
    heartbeat_file = root_path / "runtime" / "heartbeat.json"
    heartbeat_file.parent.mkdir(parents=True, exist_ok=True)
    payload = {
        "timestamp": datetime.now(timezone.utc).isoformat().replace("+00:00", "Z"),
        "status": status,
    }
    if extra:
        payload.update(extra)
    heartbeat_file.write_text(json.dumps(payload, ensure_ascii=False, indent=2), encoding="utf-8")
